### ISSUE TITLE
Function 'load_arrays' in utils.py - Unsorted file paths in 1.7.5

### ISSUE THREAD

**Issue by aleksandra-kim** [brightway-lca/brightway2-calc#25]
**[Original report](https://bitbucket.org/cmutel/brightway2-calc/issue/25) by Pedro Anchieta (Bitbucket: [phanchieta](https://bitbucket.org/phanchieta), GitHub: [phanchieta](https://github.com/phanchieta)).**

----------------------------------------

Version 1.7.4:

‌

```python
def load_arrays(paths):
    """Load the numpy arrays in list of filepaths ``paths``."""
    assert all(os.path.isfile(fp) for fp in paths)
    return np.hstack([np.load(path) for path in sorted(paths)])
```

was changed \(in version 1.7.5\) to

‌

```python
...
for obj in objs:
    if isinstance(obj, np.ndarray):
        # we're done here as the object is already a numpy array
        arrays.append(obj.copy())
    else:
        # treat object as loadable by numpy and try to load it from disk
        arrays.append(np.load(obj))
return np.hstack(arrays)
```

List "arrays" appends unsorted objects and causes errors when using file paths.

**Comment by aleksandra-kim** [comment]
**Original comment by Chris Mutel (Bitbucket: [cmutel](https://bitbucket.org/cmutel), GitHub: [cmutel](https://github.com/cmutel)).**

----------------------------------------

Thanks for the bug report! This could be a serious issue, but I am not sure I see the problem. What difference could it make if the filepaths were sorted or not?

**Comment by aleksandra-kim** [comment]
**Original comment by Pedro Anchieta (Bitbucket: [phanchieta](https://bitbucket.org/phanchieta), GitHub: [phanchieta](https://github.com/phanchieta)).**

----------------------------------------

Thanks for the reply! We implement a version of the multi-core Monte Carlo simulation and some LCA score results were incorrect when using 1.7.5 and multiple cores. Some cores were producing correct results while others were several orders of magnitude off. Making local changes to utils.py has solved the issue temporarily but other users might also come across this same problem. 

I’m unsure if this also happens in the ‘out of the box’ Monte Carlo simulation. Also, this issue only happened on Windows. Tested on Linux and results were OK. Probably due to the different file system.

**Comment by cmutel** [comment]
2.5 switches to using `bw_processing`, which will load things deterministically (not by sorting the filepaths, but by using the order provided in the metadata `datapacakge.json` file).

**Comment by cmutel** [comment]
Fixed in 2.0 development branch.
